storage/README.md

```diff
@@ -0,0 +1,130 @@
+# zipkin-storage-kafka design
+
+## Goals
+
+* Provide a fast and reliable storage that enable extensibility via Kafka topics.
+* Provide full storage functionality via streaming aggregations (e.g., spans queries, dependency graph).
+* Create a processing space where additional enrichment can be plugged-in into the tracing pipeline.
+* Remove need for additional storage when Kafka is available.
+
+## Kafka Zipkin Storage
+
+Storage is composed by 4 main components: 
+
+- Span Consumer: repartition of collected span batches into spans keyed by `traceId`
+- Span Aggregator: processing of _partitioned spans_ into aggregated traces and later into dependency links.
+- Trace and Dependency Stores: building local state stores to support search and query APIs for traces and dependencies.
+
+### Span Consumer
+
+This component processes span batches, received via HTTP, Kafka, ActiveMQ, etc.; 
+take elements with same trace-id and keyed them to be stored on `zipkin-spans` topic for repartition.
+
+| Property | Environment Variable | Description |
+|----------|----------------------|-------------|
+| `zipkin.storage.kafka.span-partitioning-enabled` | `SPAN_PARTITIONING_ENABLED` | `false` disables span partitioning. Consider that `Aggregation` component requires partitioned spans. Defaults to `true`. |
```

Comment:
prefer .enabled as that's more spring idiomatic right? I think it is but maybe not?

Reply:
ex s/partitioned-spans-topic/span-partitioning.topic

Reply:
How does this play with `*Properties` field names? I choose `span-partitioning-enabled` as field is named `spanPartitioningEnabled`, but maybe there is a tag to customize names. I'll look into it.

Agree that having .enabled, .topic will be easier to customize.

Reply:
it works by nesting classes like this https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/main/java/zipkin2/server/internal/ui/ZipkinUiProperties.java#L108